### ISSUE TITLE
[bug] `repo.firstCommitHash()` returns `undefined` - the SDK can't create project ids

### ISSUE THREAD

**Issue by samuelstroschein** [opral/monorepo#2101]
### Problem

1. The SDK can't generate project ids because
2. Lix returns `undefined` when the SDK calls `repo.firstCommitHash()`

### Expected behavior

1. The SDK generates a project_id 
2. Lix does not return `undefined` for `repo.firstCommitHash()`

### Reproduction

https://github.com/opral/monorepo/assets/35429197/7bc446b4-6e34-4e44-bb19-2b1b65c7b150

1. Start the ide extension debugger in vscode
2. Open `monorepo/inlang/source-code/end-to-end-tests/inlang-nextjs` with the vscode window that is ran as debugged version 
3. set a breakpoint in `generateProjectId()` 
<img width="601" alt="CleanShot 2024-01-24 at 11 50 37@2x" src="https://github.com/opral/monorepo/assets/35429197/a68d4e88-cad3-4ddb-bb9f-c1ada97eb7c0">

### Other information

- platform: node, mac os

**Comment by samuelstroschein** [comment]
@janfjohannes @jldec was the repro video helpful?

**Comment by samuelstroschein** [comment]
@felixhaeberle this likely means that the vscode extension is used by more projects than currently known 🚀

**Comment by jldec** [comment]
@samuelstroschein I found the video quite hard to follow without audio - in that case I'd prefer a couple of lines of text instead.

Questions: 
1. why not use a uuid for the project-id instead of depending on git.
  (if 2 users do it at the same time they'll need to resolve the merge conflict)
2. is there an option to disable telemetry?

**Comment by felixhaeberle** [comment]
> I found the video quite hard to follow without audio - in that case I'd prefer a couple of lines of text instead.

@samuelstroschein me too! please no videos without audio

**Comment by samuelstroschein** [comment]
@jldec 

1. Old projects need to be migrated with rhe same ID. Given the distributed nature, we need a common "info" from which an ID is derived. Otherwise, we will count projects multiple times and merge conflicts will arise. The first commit hash was the most realiable option. New projects can use UUIDs.

2. no, not at the moment. 


Re the video: noted. I was in a cafe when i recorded this. Recording with voice would have been rude.

**Comment by janfjohannes** [comment]
@jldec  but you can disable posthog by setting PUBLIC_POSTHOG_TOKEN=""  in case that helps.

im on it.

**Comment by janfjohannes** [comment]
@felixhaeberle  @samuelstroschein  the video feels like a draw the rest of the owl moment.  could you make a loom how to exactly do the vscode debug setup? i have no idea what to do and vscode docs also dont seem to help.

**Comment by felixhaeberle** [comment]
@janfjohannes 
https://www.loom.com/share/7cd006a145e9492ebfd0b3653f24caf2

**Comment by janfjohannes** [comment]
@felixhaeberle  thanks , your first sentence was the clou: you need to run the dev command fro the ide extension. unfortuneately this was also not enough, i needed to set 			"outFiles": []   in the luanch config, as this excluded the map file so it was not possible to use breakpoints without removing the entry there.

**Comment by felixhaeberle** [comment]
@samuelstroschein now we know why the debugging was so buggy haha

**Comment by janfjohannes** [comment]
@here i have identified the bug, it was allready a bug in my todos of fixing potential bugs.    as they say: "We have identified the issue and we are currently working on a resolution."

**Comment by felixhaeberle** [comment]
@janfjohannes If you now can also fix the always showing progress indicator on the extension icon in the sidebar I would be more than happy haha, I wasted hours on it not figuring out which processes are still running in the background.

**Comment by janfjohannes** [comment]
@felixhaeberle   sure happy to have a look, just create a new ticket and ping me there

**Comment by jldec** [comment]
from  https://github.com/opral/monorepo/issues/2101#issuecomment-1908175296
> New projects can use UUIDs.

@janfjohannes should I go ahead and push a PR loadProject to use UUID when there is no first commit? That would unblock #2069

**Comment by samuelstroschein** [comment]
> @jldec should I go ahead and push a PR loadProject to use UUID when there is no first commit? 

No. 

A project can't be loaded if no repo (first commit) exists. In #2069 I recommend throwing an error like `NoRepositoryExistsError`. Although, wait a sec. I have an idea which I thought would be too early now but it might not be!

> New projects can use UUIDs.

We will introduce a `createNewProject()` function in the SDK that will create a UUID for the newly created project.
